gsw/geostrophy.py

```diff
@@ -78,7 +78,7 @@ def geo_strf_dyn_height(SA, CT, p, p_ref=0, axis=0, max_dp=1.0,
     except AttributeError:
         order = 'C'  # e.g., xarray DataArray doesn't have flags
     for ind in indexer(SA.shape, axis, order=order):
-        igood = goodmask[ind]
+        igood = np.array(goodmask[ind])
```

Comment:
I think that using `np.asarray` would be more efficient, avoiding the copy which is unnecessary when the original function arguments are numpy arrays.  I would also add a comment that this is needed to support xarray inputs for numpy < 1.23.

Reply:
Done.